src/html.cr

```diff
@@ -36,10 +36,33 @@ module HTML
   # HTML.escape("Crystal & You", io) # => nil
   # io.to_s                          # => "Crystal &amp; You"
   # ```
+  @[AlwaysInline]
   def self.escape(string : String, io : IO) : Nil
-    string.each_char do |char|
-      io << SUBSTITUTIONS.fetch(char, char)
+    escape(string.to_slice, io)
+  end
+
+  # Same as `escape(String, IO)` but accepts `Bytes` instead of `String`.
+  #
+  # WARNING: The slice is assumed to be valid UTF-8.
+  def self.escape(string : Bytes, io : IO) : Nil
```

Comment:
I wonder if this can be generalized in `String#gsub(NamedTuple)` directly. Something like:

```crystal
def gsub(x : NamedTuple)
  gsub_named_tuple(x)
end

private def gsub_named_tuple(x : T) forall T
  {% if !T.union? && T.keys.all? { |k| k =~ /\A[[:ascii:]]\z/ && T[k] == String } %}
    # fast path
  {% else %}
    gsub do |char|
      tuple[char.to_s]? || char
    end
  {% end %}
end

gsub({"&": "&amp;", "<": "&lt;", ">": "&gt;", "\"": "&quot;", "'": "&#39;"})
```